File: contract/src/main/java/rrlane/leasing/service/VehicleService.java
package rrlane.leasing.service;

import rrlane.leasing.contract.dto.VehicleDTO;
import rrlane.leasing.entity.Vehicle;

import java.util.List;

public interface VehicleService {
    public String saveVehicle(VehicleDTO vehicleDTO);

    public VehicleDTO searchForVehicle(VehicleDTO vehicleDTO);

    public List<Vehicle> getVehiclesByDetails(String[] details);
}

File: contractapi/src/main/kotlin/de/leasing/contract/api/service/VehicleService.kt
package de.leasing.contract.api.service

import de.leasing.contract.api.entity.dto.VehicleReq
import de.leasing.contract.api.entity.dto.VehicleResp

interface VehicleService {
    fun saveVehicleRecord(customerReq: VehicleReq): VehicleResp
    fun listAllVehicleRecords(page: Int, size: Int): Iterable<VehicleResp>
    fun listAllVehicleRecords(sortBy: String?): Iterable<VehicleResp>
    fun getVehicleById(id: String): VehicleResp
}
